### PR TITLE
[Serve] [Docs] Revise Java API documentation

### PR DESCRIPTION
**Pull request by shrekris-anyscale**
## Why are these changes needed?

This change revises some wording and adds links to Serve's Java documentation.

Link to the updated documentation: https://ray--27831.org.readthedocs.build/en/27831/serve/managing-java-deployments.html

## Related issue number

N/A

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - Only documentation text is changed, so no testing is required.

### CODE REVIEW

**Review by simon-mo — APPROVED** [review]

File: doc/source/serve/managing-java-deployments.md [hunk 1]
```diff
@@ -62,7 +62,7 @@ By specifying the `numReplicas` parameter, you can change the number of deployme
 
 ### Resource Management (CPUs, GPUs)
 
-Through the `rayActorOptions` parameter, you can set the resources of deployment, such as using one GPU:
+Through the `rayActorOptions` parameter, you can reserve resources for each deployment replica, such as one GPU:
```
Comment (shrekris-anyscale) [inline doc/source/serve/managing-java-deployments.md:65]: Is this correct? Does the `rayActorOptions` parameter control resources for each replica (like the Python API), or is it for each deployment?

Comment (simon-mo) [reply doc/source/serve/managing-java-deployments.md:65]: each replica, exactly the same as python

File: doc/source/serve/managing-java-deployments.md [hunk 2]
```diff
@@ -1,52 +1,52 @@
 # Managing Java Deployments
 
-Java is one of the mainstream programming languages for production services. Ray Serve natively supports Java API for creating, updating, and managing deployments. You can create Ray Serve deployments using Java and call them via Python, or vice versa.
+Java is a mainstream programming language for production services. Ray Serve offers a native Java API for creating, updating, and managing deployments. You can create Ray Serve deployments using Java and call them via Python, or vice versa.
 
-This section helps you to:
+This section helps you:
```
Comment (dmatrix) [inline doc/source/serve/managing-java-deployments.md:6]: I would add "helps you to:"

**Review by dmatrix — APPROVED** [review]
a small nit to include a proposition. LTGM

File: doc/source/serve/managing-java-deployments.md (hunk 2 above)
Comment (shrekris-anyscale) [reply doc/source/serve/managing-java-deployments.md:6]: Sounds good, I added the `to`.

**Pull request merged**